haxe/ui/backend/ComponentBase.hx

```diff
@@ -232,7 +232,58 @@ class ComponentBase {
             window.bind(EventType.ERASE_BACKGROUND, function(e) {
 
             });
-        }        
+        }
+    }
+
+    private function replaceWindow(replacement:Window) {
+        if (replacement == null) {
+            return;
+        }
+        __children = [];
+
+        cast(this, Component).invalidateStyle(false);
+        window.destroy();
+        window = replacement;
+
+        var platform:PlatformInfo = new PlatformInfo();
```

Comment:
It depends on the plateform. According to the documentation it is possible to update the widget at runtime with wxGTK but not wxMSW.